portlayer/vsphere/storage/parent.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"path"
+	"sync"
+
+	"golang.org/x/net/context"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/soap"
+	"github.com/vmware/vic/pkg/vsphere/session"
+	"github.com/vmware/vic/pkg/vsphere/tasks"
+)
+
+const parentMFile = "parentMap"
+
+// Implements the cache used to lookup an image's parent
+type parentM struct {
+	// location in the datastore in datastore URI format
+	mFilePath string
+
+	// map of image ID to parent ID
+	db map[string]string
+
+	sess *session.Session
+
+	l sync.Mutex
+}
+
+// Starts here.  Tries to create a new parentM or load an existing one.
+func restoreParentMap(ctx context.Context, s *session.Session) (*parentM, error) {
```

Comment:
The whole file is a workaround for a vsphere API that isn't ready yet.  Once we have First Class Disk support, this can go away.  As such, keeping it private keeps this implementation from leaking into other packages.  In the end, all we need to describe a disk should be the `Image` struct.